### PR TITLE
Block Library: Update placeholder text across blocks

### PR DESCRIPTION
**Pull request by mtias**
Related #29611 and #30393.

This change updates the placeholder rich-text fields we use across blocks to favor more concise wording and remove the trailing ellipsis. It works better alongside the new persistent placeholder text until the user types.

Placeholder states are an important part of onboarding users to blocks and should encourage clarity and guidance.

### CODE REVIEW

File: packages/block-library/src/pullquote/edit.js [hunk 1]
```diff
@@ -127,7 +127,7 @@ function PullQuoteEdit( {
 						aria-label={ __( 'Pullquote text' ) }
 						placeholder={
 							// translators: placeholder text used for the quote
-							__( 'Write quote…' )
+							__( 'Add quote' )
```
Comment (mtias) [inline packages/block-library/src/pullquote/edit.js:130]: Maybe this should be just "Quote"

File: packages/block-library/src/gallery/gallery-image.js [hunk 2]
```diff
@@ -281,9 +281,7 @@ class GalleryImage extends Component {
 					<RichText
 						tagName="figcaption"
 						aria-label={ __( 'Image caption text' ) }
-						placeholder={
-							isSelected ? __( 'Write caption…' ) : null
-						}
+						placeholder={ isSelected ? __( 'Add caption' ) : null }
```
Comment (ellatrix) [inline packages/block-library/src/gallery/gallery-image.js:284]: I've not seen this pattern before. Why is the placeholder removed on deselect? Normally the whole rich caption is removed.

File: packages/block-library/src/heading/edit.js [hunk 3]
```diff
@@ -80,7 +80,7 @@ function HeadingEdit( {
 				onReplace={ onReplace }
 				onRemove={ () => onReplace( [] ) }
 				aria-label={ __( 'Heading text' ) }
-				placeholder={ placeholder || __( 'Write heading…' ) }
+				placeholder={ placeholder || __( 'Heading' ) }
```
Comment (ellatrix) [inline packages/block-library/src/heading/edit.js:83]: Why not remove "add" from other placeholders?

File: packages/block-library/src/list/edit.js [hunk 4]
```diff
@@ -155,7 +155,7 @@ export default function ListEdit( {
 				}
 				value={ values }
 				aria-label={ __( 'List text' ) }
```
Comment (ellatrix) [inline packages/block-library/src/list/edit.js:157]: I wonder why the label is different here. A comment should have been added. I think it was because the accessibility team didn't like the "Write..." prompt, which makes me wonder if we can remove it now.

File: packages/block-library/src/pullquote/edit.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-library/src/pullquote/edit.js:130]: Removing the "add" everywhere makes sense.

**Review by ellatrix — APPROVED** [review]

File: packages/block-library/src/list/edit.js (hunk 4 above)
Comment (mtias) [reply packages/block-library/src/list/edit.js:157]: Possibly! I'm not yet sure if we should go with the plain text or include something like "Add List"

File: packages/block-library/src/heading/edit.js (hunk 3 above)
Comment (mtias) [reply packages/block-library/src/heading/edit.js:83]: I wanted to keep it on fields like caption or citation, which are optional. The Quote is the odd one out at the moment which we should probably align

File: packages/block-library/src/gallery/gallery-image.js (hunk 2 above)
Comment (mtias) [reply packages/block-library/src/gallery/gallery-image.js:284]: I don't know, but it will likely disappear when the gallery refactor lands

Comment (ellatrix) [reply packages/block-library/src/gallery/gallery-image.js:284]: Yes, just noticing it, no big deal. :)

**Review by aristath — APPROVED** [review]
